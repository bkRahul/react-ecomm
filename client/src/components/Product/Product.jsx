import React from 'react';
import { connect } from 'react-redux';

import classes from './Product.module.scss';
import { addToCart } from '../../redux/store/cart/cart.actions';
import { Button } from '../../ui/Button/Button';

const Product = ({ item, addToCart }) => {
  const { name, imageUrl, price } = item;
  return (
    <div className={classes.Product}>
      <div
        className={classes.Image}
        style={{ backgroundImage: `url(${imageUrl})` }}
      >
        <Button btnType="ProductBtn" clicked={() => addToCart(item)}>
          add to cart
        </Button>
      </div>
      <div className={classes.CollectionFooter}>
        <span className={classes.Name}>{name}</span>
        <span className={classes.Price}>{price}</span>
      </div>
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  addToCart: item => dispatch(addToCart(item)),
});

export default connect(null, mapDispatchToProps)(Product);
