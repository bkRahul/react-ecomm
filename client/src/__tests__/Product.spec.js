import React from 'react'
import { shallow, render, mount } from 'enzyme'
import { Product } from '../components/Product/Product'

describe('Product Component', () => {
	let wrapper
	let mockAddToCart
	let mockProps

	beforeEach(() => {
		mockAddToCart = jest.fn()
		mockProps = {
			item: {
				imageUrl: 'www.testImage.com',
				price: 10,
				name: 'Black hat',
			},
			addToCart: mockAddToCart,
		}

		wrapper = shallow(<Product {...mockProps} />)
	})

	it('to render a render correctly', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('should render with Product Name', () => {
		expect(wrapper.find('span').first().text()).toEqual(mockProps.item.name)
	})

	it('should render with Product render Price', () => {
		const price = parseInt(wrapper.find('[id="price"]').text())
		expect(price).toEqual(mockProps.item.price)
	})
})

describe('Product Component with Button', () => {
	let wrapper
	let mockAddToCart
	let mockProps

	beforeEach(() => {
		mockAddToCart = jest.fn()
		mockProps = {
			item: {
				imageUrl: 'www.testImage.com',
				price: 10,
				name: 'Black hat',
			},
			addToCart: mockAddToCart,
		}

		wrapper = mount(<Product {...mockProps} />)
	})

	it('to render a correctly', () => {
		expect(wrapper.find('button').text()).toEqual('add to cart')
	})

	it(' to call addToCart when add to cart button is clicked', () => {
		wrapper.find('button').simulate('click')
		expect(mockAddToCart).toHaveBeenCalled()
	})
})
