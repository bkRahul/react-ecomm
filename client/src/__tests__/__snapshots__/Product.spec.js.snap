// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component to render a render correctly 1`] = `ShallowWrapper {}`;
